### PR TITLE
Class features loose should have precedence over preset-env

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes https://github.com/babel/babel/issues/11622
| Patch: Bug Fix?          | y
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Please suggest me a better method to fix this if possible 😭

### CODE REVIEW

File: packages/babel-helper-create-class-features-plugin/src/features.js [hunk 1]
```diff
@@ -18,49 +24,67 @@ export const FEATURES = Object.freeze({
...
   //   @babel/plugin-class-properties { loose: false }
   // is transformed in loose mode.
   // We only enabled the feature if it was previously disabled.
-  if (!hasFeature(file, feature)) {
+  if (!hasFeature(file, feature) || canIgnoreLoose(file, feature)) {
     file.set(featuresKey, file.get(featuresKey) | feature);
-    if (loose) file.set(looseKey, file.get(looseKey) | feature);
+    if (
+      loose ===
+      "#__internal__@babel/preset-env__prefer-true-but-false-is-ok-if-it-prevents-an-error"
```
Comment (JLHwung) [inline packages/babel-helper-create-class-features-plugin/src/features.js:49]: Lol I can't prevent myself from thinking of https://github.com/facebook/react/blame/80bff5397bf854750dbe7c286f61654ea58938c5/src/umd/ReactUMDEntry.js#L21

File: packages/babel-helper-create-class-features-plugin/src/features.js [hunk 2]
```diff
@@ -18,49 +24,67 @@ export const FEATURES = Object.freeze({
...
+
+    const loose = isLoose(file, mask);
+
+    if (canIgnoreLoose(file, mask)) {
+      continue;
+    } else if (resolvedLoose === !loose) {
+      throw new Error(
+        "'loose' mode configuration must be the same for @babel/plugin-proposal-class-properties, " +
+          "@babel/plugin-proposal-private-methods and " +
+          "@babel/plugin-proposal-private-property-in-object (whe they are enabled).",
```
Comment (JLHwung) [inline packages/babel-helper-create-class-features-plugin/src/features.js:77]:
```suggestion
          "@babel/plugin-proposal-private-property-in-object (when they are enabled).",
```

File: packages/babel-preset-env/src/index.js [hunk 3]
```diff
@@ -301,10 +301,25 @@ export default declare((api, opts) => {
...
   const pluginUseBuiltIns = useBuiltIns !== false;
   const plugins = Array.from(pluginNames)
-    .map(pluginName => [
-      getPlugin(pluginName),
-      { spec, loose, useBuiltIns: pluginUseBuiltIns },
-    ])
+    .map(pluginName => {
+      if (
+        pluginName === "proposal-class-properties" ||
+        pluginName === "proposal-private-methods"
```
Comment (JLHwung) [inline packages/babel-preset-env/src/index.js:307]: I think we can use `classFeaturePlugins = ["proposal-class-properties", "proposal-private-methods", "proposal-private-property-in-object"]` to align with the class feature plugin factory. Although it is not included in preset-env yet but we are likely to forget adding them when `privateIn` reaches stage 3.

**Comment by existentialism** [comment]
Thought about this a bit... I think this is fine. Can we possibly emit a warning when the loose mode from `@babel/preset-env` is overridden by another plugin's option?

**Comment by nicolo-ribaudo** [comment]
@existentialism I have added the warning. You can see in the tests added in the last commit that it can still be explicitly silenced by explicitly enabling the plugins with the correct loose mode.

**Comment by existentialism** [comment]
Is it possible to show them which plugin overrode?

I think something like:

> Though the `loose` option was set to `true` in your `@babel/preset-env` config, it will not be used since the 'loose' mode option was set for {plugin name}. The `loose` option must be the same for @babel/plugin-proposal-class-properties, @babel/plugin-proposal-private-methods and @babel/plugin-proposal-private-property-in-object (when they are enabled).

Might be slightly more helpful in both letting them know what is going on, and how to fix it.

**Comment by nicolo-ribaudo** [comment]
Better?

**Comment by existentialism** [comment]
I think it is, you?

We might be able to shorten it, but thats more bikeshedding for later. I think it describes the problem well and how to fix it.

**Comment by nicolo-ribaudo** [comment]
Yeah, we don't need to shorten it. It's more important to exactly explain what's going on and how to fix it.

**Pull request merged**